src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -6003,6 +6003,25 @@ void AuraEffect::HandleStoreTeleportReturnPoint(AuraApplication const* aurApp, u
         playerTarget->RemoveStoredAuraTeleportLocation(GetSpellInfo()->Id);
 }
 
+void AuraEffect::HandleCosmeticMounted(AuraApplication const* aurApp, uint8 mode, bool apply) const
+{
+    if (!(mode & AURA_EFFECT_HANDLE_REAL))
+        return;
+
+    if (apply)
+    {
+        aurApp->GetTarget()->Dismount();
+        aurApp->GetTarget()->SetCosmeticMountDisplayId(GetMiscValue());
+    }
+    else
+        aurApp->GetTarget()->SetCosmeticMountDisplayId(0);
```

Comment:
This is correct, even if you have two auras of this type (tested zandalari racial + divine steed, got set to 0 after steed expired), please document it